<launch>
  <arg name="load_manager" />
  <arg name="manager_name" />
  <arg name="manager_threads" />
  <arg name="camera_name" />

  <node if="$(arg load_manager)" pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen">
    <param name="num_worker_threads" value="$(arg manager_threads)"/>
  </node>

  <!-- De-Bayer / rectify -->
  <include file="$(find image_proc)/launch/image_proc.launch">
    <arg name="manager" value="$(arg manager_name)"/>
  </include>

  <!-- Apriltag detection -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag" output="screen">
    <param name="camera_frame" type="str" value="$(arg camera_name)"/>
    <param name="publish_tag_detections_image" type="bool" value="true"/>
    <rosparam command="load" file="$(find camera_tracking)/config/apriltag_settings.yaml"/>
    <rosparam command="load" file="$(find camera_tracking)/config/tags.yaml"/>
  </node>

  <!-- Localization -->
  <node pkg="camera_tracking" type="localizer" name="localizer" output="screen">
    <param name="origin_id" type="str" value="1"/>
    <param name="buffer_size" type="int" value="100"/>
    <rosparam param="pub_topics">
      '1': /origin/pose
      '69': /car0/pose
    </rosparam>
  </node>
</launch>
